`or` condition works incorrectly for missing/disappeared elements

Give that `missing` element doesn't exist,
Usual "disappear" check for normally for non-existing elements:

```java
$("#missing").should(disappear);
```


But "or" condition doesn't work:

```java
$("#missing").should(disappear.or(text("I don't exist")));
```

Expected behaviour: "or" method doesn't throw any exceptions.
Actual behaviour: it throws an exception:

```java
Element not found {#missing}
Expected: be hidden OR text "I don't exist"
...
Caused by: NoSuchElementException: no such element: Unable to locate element: {"method":"css selector","selector":"#missing"}
```

--- a/src/main/java/com/codeborne/selenide/WebElementCondition.java
+++ b/src/main/java/com/codeborne/selenide/WebElementCondition.java
@@ -61,7 +61,11 @@ public abstract class WebElementCondition {
    * @see <a href="https://github.com/selenide/selenide/wiki/do-not-use-getters-in-tests">NOT RECOMMENDED</a>
    */
   public WebElementCondition or(WebElementCondition alternative) {
-    return new WebElementCondition("%s OR %s".formatted(WebElementCondition.this.toString(), alternative.toString())) {
+    String conditionName = "%s OR %s".formatted(WebElementCondition.this.toString(), alternative.toString());
+    boolean acceptsMissingElements =
+      WebElementCondition.this.missingElementSatisfiesCondition() || alternative.missingElementSatisfiesCondition();
+
+    return new WebElementCondition(conditionName, acceptsMissingElements) {
       @Override
       public CheckResult check(Driver driver, WebElement element) {
         CheckResult r1 = WebElementCondition.this.check(driver, element);

--- a/src/test/java/integration/LogicalOperationsWithConditions.java
+++ b/src/test/java/integration/LogicalOperationsWithConditions.java
@@ -10,6 +10,7 @@ import java.time.Duration;
 
 import static com.codeborne.selenide.Condition.and;
 import static com.codeborne.selenide.Condition.be;
+import static com.codeborne.selenide.Condition.disappear;
 import static com.codeborne.selenide.Condition.exist;
 import static com.codeborne.selenide.Condition.not;
 import static com.codeborne.selenide.Condition.or;
@@ -67,6 +68,13 @@ final class LogicalOperationsWithConditions extends ITest {
     $("#remove").shouldBe(visible.or(exist));
     $("#remove").shouldBe(text("Remove me").or(text("Delete me")));
     $("#remove").shouldBe(text("Delete me").or(text("Remove me")));
+  }
+
+  @Test
+  void or_newerSyntax_missingElement() {
+    $("#missing").should(disappear);
+    $("#missing").should(disappear.or(text("I don't exist")));
+    $("#missing").shouldHave(text("I don't exist").or(text("Not today")).or(disappear));
   }
 
   @Test

--- a/statics/src/test/java/integration/OfflineModeTest.java
+++ b/statics/src/test/java/integration/OfflineModeTest.java
@@ -1,7 +1,9 @@
 package integration;
 
+import com.codeborne.selenide.WebElementCondition;
 import org.junit.jupiter.api.BeforeEach;
 import org.junit.jupiter.api.Test;
+import org.openqa.selenium.By;
 import org.openqa.selenium.chromium.ChromiumNetworkConditions;
 import org.openqa.selenium.chromium.HasNetworkConditions;
 import org.openqa.selenium.devtools.DevTools;
@@ -23,8 +25,17 @@ final class OfflineModeTest extends ITest {
   private static final List<NetworkConditions> ALL_REQUESTS =
     List.of(new NetworkConditions("", 0, 0, 0, empty(), empty(), empty(), empty()));
 
+  private static final WebElementCondition NO_INTERNET =
+    text("no internet")
+      .or(text("not connected"))
+      .or(text("Press space to play"))
+      .or(disappear);
+
+  private final By header = By.cssSelector("h1");
+
   @BeforeEach
   void openTestPage() {
+    setTimeout(1_000);
     openFile("file_upload_form.html");
   }
 
@@ -36,16 +47,16 @@ final class OfflineModeTest extends ITest {
     networkConditions.setOffline(true);
     HasNetworkConditions networkConditionsDriver = (HasNetworkConditions) driver().getWebDriver();
 
-    $("h1").shouldHave(text("File upload form"));
+    $(header).shouldHave(text("File upload form"));
 
     networkConditionsDriver.setNetworkConditions(networkConditions);
     driver().refresh();
 
-    $("h1").shouldHave(text("no internet").or(text("not connected")).or(disappear));
+    $(header).shouldHave(NO_INTERNET);
 
     networkConditionsDriver.setNetworkConditions(new ChromiumNetworkConditions());
     driver().refresh();
-    $("h1").shouldHave(text("File upload form"));
+    $(header).shouldHave(text("File upload form"));
   }
 
   @Test
@@ -59,18 +70,18 @@ final class OfflineModeTest extends ITest {
     // Enable network emulation
     devTools.send(Network.enable(empty(), empty(), empty(), empty(), empty()));
 
-    $("h1").shouldHave(text("File upload form"));
+    $(header).shouldHave(text("File upload form"));
 
     // Go offline
     devTools.send(Network.emulateNetworkConditionsByRule(true, ALL_REQUESTS));
 
     driver().refresh();
 
-    $("h1").shouldHave(text("no internet").or(text("not connected")).or(disappear));
+    $(header).shouldHave(NO_INTERNET);
 
     // Bring back online
     devTools.send(Network.emulateNetworkConditionsByRule(false, ALL_REQUESTS));
     driver().refresh();
-    $("h1").shouldHave(text("File upload form"));
+    $(header).shouldHave(text("File upload form"));
   }
 }

